File: core/src/com/uw/exception/CollisionException.java
package com.uw.exception;

import com.uw.object.unplayable.BasicGltfObject;

import java.util.Set;
import java.util.stream.Collectors;

public class CollisionException extends RuntimeException {
    public CollisionException(Set<BasicGltfObject> intersections) {
        super("User stuck in object: " + intersections.stream().map(BasicGltfObject::toString).collect(Collectors.joining(", ")));
    }
}
    

File: core/src/com/uw/exception/CollisionException.kt
package com.uw.exception

import com.uw.`object`.unplayable.BasicObject
import java.util.stream.Collectors

class CollisionException(intersections: Set<BasicObject>) :
    RuntimeException(
        "User stuck in object: " + intersections.stream().map { obj: BasicObject -> obj.toString() }
            .collect(Collectors.joining(", "))
    )
